ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/BlockBody.java

```diff
@@ -75,34 +90,36 @@ public static BlockBody readFrom(
     final BlockBody body =
         new BlockBody(
             input.readList(Transaction::readFrom),
-            input.readList(rlp -> BlockHeader.readFrom(rlp, blockHeaderFunctions)));
+            input.readList(rlp -> BlockHeader.readFrom(rlp, blockHeaderFunctions)),
+            Optional.empty());
     input.leaveList();
     return body;
   }
 
   @Override
-  public boolean equals(final Object obj) {
-    if (obj == this) {
-      return true;
-    }
-    if (!(obj instanceof BlockBody)) {
-      return false;
-    }
-    final BlockBody other = (BlockBody) obj;
-    return transactions.equals(other.transactions) && ommers.equals(other.ommers);
+  public boolean equals(final Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    BlockBody blockBody = (BlockBody) o;
+    return Objects.equals(transactions, blockBody.transactions)
+        && Objects.equals(ommers, blockBody.ommers)
+        && Objects.equals(withdrawals, blockBody.withdrawals);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(transactions, ommers);
+    return Objects.hash(transactions, ommers, withdrawals);
   }
 
   @Override
   public String toString() {
-    final StringBuilder sb = new StringBuilder();
-    sb.append("BlockBody{");
-    sb.append("transactions=").append(transactions).append(", ");
-    sb.append("ommers=").append(ommers);
-    return sb.append("}").toString();
+    return "BlockBody{"
+        + "transactions="
+        + transactions
+        + ", ommers="
+        + ommers
+        + ", withdrawals="
+        + withdrawals
```

Comment:
Will this just return ", withdrawals=Optional.empty" ?

Reply:
Yeah that's what it will return